tests/main_test.go

```diff
@@ -65,6 +66,8 @@ func TestMain(m *testing.M) {
 	clientSet, config := matrix.NewClientSet()
 	manager = matrix.NewKubeManager(clientSet, config)
 	namespace = matrix.GetNamespace()
+	sonobuoyResultsWriter := pluginhelper.NewDefaultSonobuoyResultsWriter()
+	progressReporter := pluginhelper.NewProgressReporter(5)
```

Comment:
Currently there is no way to get a test count without hardcoding it. Up to you how to proceed here. I added the 5 manually; but it is a pain to keep updating that.

I have a ticket in sonobuoy to show better output when we dont know the number of tests, but currently it just shows things like `3/0 (1 failure)` which could be a bit confusing to someone.

Reply:
I see. kinda wonder where 5 comes from? is it the number of test entries to be reported

Reply:
Yes, it just happens to be the number of times the `AfterEachTest` gets invoked which corresponds to the # of tests.

Reply:
Ideally, the e2e framework would have a way to find the number of tests that _will be_ run. This is what ginkgo does and before the suite starts we grab that number in a hook. There is a ticket currently for the e2e framework to do something similar.

Reply:
Sure we can remove this hard coded value once e2e-framework is updated with test counter

Reply:
https://github.com/kubernetes-sigs/e2e-framework/issues/62

Reply:
that's great, the current number should be 12

Reply:
So that is probably because of the difference between the `afterEachTest` and `afterEachFeature`. I can try it out with the other hook and see how it goes.

Reply:
that would be great to add, thanks John！

Reply:
So I just made https://github.com/kubernetes-sigs/e2e-framework/issues/101

The afterEachFeature doesn't have pass/fail info. It seems like maybe there would need to be some refactoring to pull those features apart rather than tested in the same "TestFoo"?

Is there a larger reason why the features were defined this way? For instance, `TestBasicService` defines numerous features and then runs
```
	testenv.Test(t, featureClusterIP, featureNodePort, featureLoadBalancer, featureEndlessService, featureHairpin, featureSessionAffinity)

```